src/coreclr/tools/aot/ILCompiler.ReadyToRun/JitInterface/CorInfoImpl.ReadyToRun.cs

```diff
@@ -2206,11 +2206,7 @@ private void embedGenericHandle(ref CORINFO_RESOLVED_TOKEN pResolvedToken, bool
                             MethodDesc md = HandleToObject(pResolvedToken.hMethod);
                             TypeDesc td = HandleToObject(pResolvedToken.hClass);
 
-                            bool unboxingStub = false;
-                            if ((td.IsValueType) && !md.Signature.IsStatic)
-                            {
-                                unboxingStub = true;
-                            }
+                            bool unboxingStub = md.IsUnboxingThunk();
```

Comment:
I have tried to mimic it from cg1: https://github.com/dotnet/runtime/blob/01b7e73cd378145264a7cb7a09365b41ed42b240/src/coreclr/vm/zapsig.cpp#L1257. But not sure if `UnboxingThunk` and `UnboxingStub` mean the same thing? 

I have validated the repro test works with this change, but not sure if it breaks other scenarios.

Reply:
So this test caught a regression: https://github.com/dotnet/runtime/blob/49aec59e52d2f0bc8e2acd297023aa8464aee00c/src/tests/Regressions/coreclr/GitHub_41674/genericldtoken.il#L35

It appears that this flag is not correctly propagated in crossgen2: https://github.com/dotnet/runtime/blob/dec321c12ed283b3c19e829ed7d7a2248ce072b0/src/coreclr/vm/method.hpp#L1837

Needs some better handling here, @davidwrighton in case you have thoughts on this.  

cg1:
```
    TableIndex 2, Offset 0002: [UNBOX] string GenericStruct`1<string>.B() (METHOD_HANDLE)
```
cg2:
```
    TableIndex 4, Offset 0000: [INST] string GenericStruct`1<string>.B() (METHOD_HANDLE)
```

Reply:
Ok a more reasonable fix is to check if the method is `!Static && IsVirtual`. The runtime behavior is here: https://github.com/dotnet/runtime/blob/01b7e73cd378145264a7cb7a09365b41ed42b240/src/coreclr/vm/methodtablebuilder.cpp#L6984. For now just added the virtual check but need to determine if the other two checks need to be handled as well.